ansi/table.go

```diff
@@ -32,76 +34,86 @@ type TableCellElement struct {
 func (e *TableElement) Render(w io.Writer, ctx RenderContext) error {
 	bs := ctx.blockStack
 
-	var indentation uint
-	var margin uint
 	rules := ctx.options.Styles.Table
-	if rules.Indent != nil {
-		indentation = *rules.Indent
-	}
-	if rules.Margin != nil {
-		margin = *rules.Margin
-	}
-
-	iw := indent.NewWriterPipe(w, indentation+margin, func(wr io.Writer) {
-		renderText(w, ctx.options.ColorProfile, bs.Current().Style.StylePrimitive, " ")
-	})
-
 	style := bs.With(rules.StylePrimitive)
-	ctx.table.styleWriter = NewStyleWriter(ctx, iw, style)
 
 	renderText(w, ctx.options.ColorProfile, bs.Current().Style.StylePrimitive, rules.BlockPrefix)
-	renderText(ctx.table.styleWriter, ctx.options.ColorProfile, style, rules.Prefix)
-	ctx.table.writer = tablewriter.NewWriter(ctx.table.styleWriter)
+	renderText(w, ctx.options.ColorProfile, style, rules.Prefix)
+	ctx.table.lipgloss = table.New().StyleFunc(func(row, col int) lipgloss.Style { return cellStyle })
```

Comment:
I've tested the PR on some of the large tables that I'd like to display with it and it already looks a lot better! However, I noticed that the width is not properly set, this does the trick:

```suggestion
	ctx.table.lipgloss = table.New().
		StyleFunc(func(row, col int) lipgloss.Style { return cellStyle }).
		Width(int(ctx.blockStack.Width(ctx)))
```

The downside is that it causes columns which contain rows with long text[^1] to be truncated or even omitted entirely. I'd prefer if it either wraps text within the cell or the table is printed in full width (which isn't really possible because it seems like glamour cuts off the lines, trying to wrap them?). But this is probably a feature request towards lipgloss and not glamour.

[^1]: If there's 100 empty rows and one with 100 characters the column is more likely to get truncated as explained [here](https://github.com/charmbracelet/lipgloss/blob/29fafad5be923707318d4557fb39e80ffc0fa6ae/table/table.go#L252-L286).